lib/rouge/lexers/powershell.rb

```diff
@@ -3,676 +3,220 @@
+        rule %r/"+/, Str::Heredoc
+        mixin :variable
+      end
 
-        rule %r/\*/, Keyword
+      state :class do
+        rule %r/\{/, Punctuation, :pop!
+        rule %r/\s+/, Text::Whitespace
+        rule %r/\w+/, Name::Class
+        rule %r/[:,]/, Punctuation
+      end
 
-        rule %r/;/, Text
-        rule %r/[^=\*\s{}()$"'`<]+/, Text
-        rule %r/\d+(?= |\Z)/, Num
-        rule %r/</, Text
-        mixin :interp
+      state :hasht do
+        rule %r/\s+/, Text::Whitespace
+        rule %r/\}/, Punctuation, :pop!
+        rule %r/"/, Str::Double, :dq
+        rule %r/'/, Str::Single, :sq
+        rule %r/\w+/, Name::Other
+        rule %r/=/, Operator
+        rule %r/,/, Punctuation
+        mixin :variable
       end
 
-      prepend :basic do
-        rule %r(<#[\s\S]*?#>)m, Comment::Multiline
-        rule %r/#.*$/, Comment::Single
-        rule %r/\b(#{OPERATORS})\s*\b/i, Operator
-        rule %r/\b(#{ATTRIBUTES})\s*\b/i, Name::Attribute
-        rule %r/\b(#{KEYWORDS})\s*\b/i, Keyword
-        rule %r/\b(#{KEYWORDS_TYPE})\s*\b/i, Keyword::Type
-        rule %r/\bcase\b/, Keyword, :case
-        rule %r/\b(#{BUILTINS})\s*\b(?!\.)/i, Name::Builtin
+      state :array do
+        rule %r/\s+/, Text::Whitespace
+        rule %r/\)/, Punctuation, :pop!
+        rule %r/"/, Str::Double, :dq
+        rule %r/'/, Str::Single, :sq
+        rule %r/,/, Punctuation
+        mixin :variable
+      end
+
+      state :bracket do
+        rule %r/\]/, Punctuation, :pop!
+        rule %r/[A-Za-z]\w+\./, Name::Constant
+        rule %r/([A-Za-z]\w+)/ do |m|
+          if ATTRIBUTES.include? m[0]
+            token Name::Builtin::Pseudo
+          else
+            token Keyword::Type
+          end
+        end
+        mixin :root
+      end
+
+      state :parameters do
+        rule %r/\n/, Text::Whitespace, :pop!
+        rule %r/[;(){}\]]/, Punctuation, :pop!
+        rule %r/[|=]/, Operator, :pop!
+        rule %r/\w[-\w]+/, Name::Other
+        mixin :root
+      end
+
+      state :root do
+        rule %r/\s+/, Text::Whitespace
+
+        rule %r/#requires\s-version \d(?:\.\d+)?/, Comment::Preproc
+        rule %r/#.*/, Comment
+        rule %r/<#/, Comment::Multiline, :multiline
+
+        rule %r/"/, Str::Double, :dq
+        rule %r/'/, Str::Single, :sq
+        rule %r/@"/, Str::Heredoc, :heredoc
+        rule %r/@'.*?'@/m, Str::Heredoc
+
+        rule %r/\d*\.\d+/, Num::Float
+        rule %r/\d+/, Num::Integer
+
+        rule %r/\.\.(?=\.?\d)/, Operator
+        rule %r/(?:#{OPERATORS})\b/i, Operator
+
+        rule %r/(class)(\s+)(\w+)/i do
+          groups Keyword::Reserved, Text::Whitespace, Name::Class
+          push :class
+        end
+        rule %r/(function)(\s+)(?:(\w+)(:))?(\w[-\w]+)/i do
+          groups Keyword::Reserved, Text::Whitespace, Name::Namespace, Punctuation, Name::Function
+        end
+        rule %r/(?:#{AUTO_VARS})\b/i, Name::Builtin::Pseudo
```

Comment:
@pyrmont  I just want to re-iterate that I think this is an over-simplistic way of identifying the automatic variables -- I did it as a quick test locally, but it's not going to correctly highlight them in strings and possibly in other contexts.

Reply:
Ah. Good to know. I should probably move the test to the `:variable` state.